api/data_explorer/controllers/export_url_controller.py

```diff
@@ -285,9 +282,9 @@ def _get_filter_query(filters):
 
     table_columns = dict()
     for filter_str in filters:
-        facet, value = _get_facet_and_value(filter_str)
+        es_field_name, facet, value = _get_facet_and_value(filter_str)
```

Comment:
nit: I might change `es_field_name` to `facet_name`. In the case of the Samples Overview facet, this isn't actually an ES field name, and we may end up having more `FiltersFacets` in the future, where there isn't a single ES field.

Reply:
Hmm I prefer `es_field_name`. I'd rather the code apply to the normal, non-Samples Overview case, I think it's easier to read.

Reply:
I'd argue that making it easier to read is actually misrepresenting what it is. You can't directly use this as an elastic search field name, because it won't always be one. I'd find it confusing  if that fact were not obvious, and obscuring that makes the code misleading. 

In locations where we are directly extracting `es_field_name` from the facet_info dict, I think it is correct to call it `es_field_name`. Eg:
```
es_field_name = current_app.config['UI_FACETS'][facet_name].get('es_field_name')
```
That field will not be present for the Samples Overview so it is not misrepresenting anything.

Reply:
We shouldn't use `facet_name` because that could be "Facet UI name". So either `es_field_name` or `facet_key`.

So right now we have:
- Facet UI name
- Elasticsearch field name
- Elasticsearch facet object

The problem with `facet_key` is, it's adding another concept that everyone has to understand. And when people write code, they'll have to think "Should this be `facet_key` or `es_field_name`?". And it's going to be impossible to enforce this 100% -- there is going to be `es_field_name` in some places where technically it should be `facet_key`.

Also, every time we use `facet_key` we would have to document "This is es_field_name except for Samples Overview". For example, in api.yaml. While the `facet_key` approach is technically more correct, I think from a pragmatic point of view, it makes the code base harder to understand and maintain.

But if you feel strongly, we can do `facet_key`. :)

Reply:
I think we absolutely **should** document everywhere that it is an es_field_name except for samples overview, regardless of which name we end up choosing. Otherwise it is misleading.

I do still prefer for `facet_key` or `facet_id` as I think it is overall more maintainable to define a new concept rather than overload an existing one and introduce caveats. Say, for example, that we decided to construct the ES facets in the controllers rather than in `__main__` (something we could think about). We cannot simply pass in the key for the facet as the elastic search field, we would need special handling to determine how to construct the particular facet and what the relevant elastic search field(s) are. Additionally, if we allowed people to specify `FiltersFacets` in `ui.json` those would all have the same structure as samples overview and would have to be special cased again. I think right now it might be a little easier to read if it were `es_field_name`, but we should design the API to support these types of things more generally rather than writing in specific code. This is similar to how I initially special cased samples as the only nested facet, but then realized that it's a more general need.

Reply:
SG. How about you do this in a separate PR, since this is a big change, and might (?) involve things outside this PR. I like `facet_id` better than `facet_key`.

Reply:
SGTM, I'll updated after this goes in.